plugins/ltac/g_auto.mlg

```diff
@@ -116,12 +116,20 @@ END
 
 let make_depth n = snd (Eauto.make_dimension n None)
 
+(* deprecated in 8.13; the second int_or_var will be removed *)
+let deprecated_eauto_bfs =
+  CWarnings.create
+    ~name:"eauto_bfs" ~category:"deprecated"
+    (fun () -> Pp.str "The syntax [eauto @int_or_var @int_or_var] is deprecated. Use [bfs eauto] instead.")
+
 }
 
 TACTIC EXTEND eauto
 | [ "eauto" int_or_var_opt(n) int_or_var_opt(p) auto_using(lems)
     hintbases(db) ] ->
-    { Eauto.gen_eauto (Eauto.make_dimension n p) (eval_uconstrs ist lems) db }
+    {
+    if n <> None && p <> None then deprecated_eauto_bfs ();
```

Comment:
As you may have heard, polymorphic equality is frowned upon in this development team, use `match` instead:
```suggestion
    match n, p with Some _, Some _ -> deprecated_eauto_bfs () | _ -> ();
```

Reply:
I saw one brief mention but not a good explanation.  It seems excessive to avoid using `=` and `<>` when the types are known, e.g. one of the values is a constant, a very common case.  The only example I can think of when you might run into this is with a match clause like `| a, b when a = b -> ...` which seems like it would be infrequent.